internal/pkg/types/signed_message.go

```diff
@@ -110,8 +110,17 @@ func (smsg *SignedMessage) VerifySignature() error {
 }
 
 // OnChainLen returns the amount of bytes used to represent the message on chain.
+// Dragons: verify this is the correct way to determine a signed messages size on chain.
```

Comment:
I am unsure the below implementation is correct, but need this method to not panic else it borks the chain-validation tests.

Reply:
Just encode and measure the size of the whole message. Remove the dragons, but leave a todo that we could save this redundant encoding if we plumbed the size through from when the message was originally decoded from the network.